src/app/load-app.js

```diff
@@ -9,37 +9,42 @@ import css from './load-app.module.css'
 const LoadApp = ({ children }) => {
     useCustomFormsPrefetch()
 
-    const { isLoading, isError, data, error } = useMetadata()
-    const {
-        isLoading: userLoading,
-        isError: userIsError,
-        data: userData,
-    } = useUserInfo()
-
-    if (isLoading || userLoading) {
+    const metadata = useMetadata()
+    const userInfo = useUserInfo()
+
+    if (metadata.isLoading || userInfo.isLoading) {
         return (
             <CenteredContent className={css.center} position={'top'}>
                 <CircularLoader />
             </CenteredContent>
         )
     }
 
-    if (isError) {
-        console.error(data)
+    if (metadata.isError) {
         return (
             <NoticeBox
-                className={css.noticeBoxWrapper}
                 error
+                className={css.noticeBoxWrapper}
                 title={i18n.t('There was a problem loading metadata')}
             >
-                {error?.message}
+                {metadata.error}
             </NoticeBox>
         )
     }
 
-    if (data && (userData || userIsError)) {
-        return children
+    if (!metadata.data) {
+        return (
+            <NoticeBox
+                error
+                className={css.noticeBoxWrapper}
+                title={i18n.t('There was a problem loading metadata')}
+            >
+                {i18n.t('The metadata response is invalid')}
+            </NoticeBox>
+        )
```

Comment:
Why is this here? Does this actually happen? I would have expected that if `metadata.isLoading` and `metadata.isError` are both `false`, that `metadata.data` would always be populated.

Reply:
What I observed in the code prior to my changes is:

* There is no explicit condition for when this is the case, but there's a check for the existence of `data` (`metadata.data` after I changed the code)
* Children won't be rendered if there's no user data or an error.

I suppose in regards to the user info we could add a case similar to the one I added for missing metadata data for the userData as well, because right now there would be just a blank workspace if that happens

Reply:
@HendrikThePendric and I have agreed that it doesn't make much sense to check for the presence of `.data`, so now all loading and error states have been covered instead